### ISSUE TITLE
Cannot use particular version of openssl

### ISSUE THREAD

**Issue by g10guang** [curl/curl#2780]
### I did this

I try to build curl from source.

```
./configure --with-ssl=/data/elioliu/localopenssl/ --prefix=$HOME/localcurl
```

### return

```
configure: Configured to build curl/libcurl:

  curl version:    7.19.5-CVS
  Host setup:      i686-suse-linux
  Install prefix:  /data/elioliu/localcurl
  Compiler:        gcc
       (--with-ssl / --with-gnutls)
  SSH support:     no      (--with-libssh2)
  zlib support:    enabled
  krb4 support:    no      (--with-krb4*)
  GSSAPI support:  no      (--with-gssapi)
  SPNEGO support:  no      (--with-spnego)
  c-ares support:  no      (--enable-ares)
  ipv6 support:    no      (--enable-ipv6)
  IDN support:     no      (--with-libidn)
  Build libcurl:   Shared=yes, Static=yes
  Built-in manual: enabled
  Verbose errors:  enabled (--disable-verbose)
  SSPI support:    no      (--enable-sspi)
  ca cert bundle:  no
  ca cert path:    no
  LDAP support:    no      (--enable-ldap / --with-ldap-lib / --with-lber-lib)
  LDAPS support:   no      (--enable-ldaps)
```

Why ` SSL support:     no` ?

### I expected the following

Use a particular openssl version.

### curl/libcurl version

```
elioliu@DevNet_34_2_sles10sp1:~/localcurl/bin> ./curl --version
curl 7.19.5-CVS (i686-suse-linux) libcurl/7.19.5-CVS zlib/1.2.3
Protocols: tftp ftp telnet dict http file 
Features: Largefile libz
```

### operating system

```
elioliu@DevNet_34_2_sles10sp1:~/code/test/libcurl> uname -a
Linux DevNet_34_2_sles10sp1 2.6.16.60-0.21-domU-101126 #1 SMP Fri Nov 26 11:56:53 CST 2010 i686 i686 i386 GNU/Linux

```

**Comment by g10guang** [comment]
@dfandrich Thank you.

I figure out what is going on. Maybe configure will analyse the openssl directory. I just pull openssl release code from github and make it, but without `make install`. I guess configure analyse the directory structure. So it think it's not a directory for openssl.

After I run `make install` for openssl. Everything goes well.😊

**Comment by g10guang** [comment]
But I think configure should recogonize the directory for openssl which without run `make install`.

**Comment by jzakrzewski** [comment]
This is what I always complain about with the configure script: if I say '--with-ssl=...' I expect a hard error if it cannot detect valid OpenSSL installation in the given directory. Not some single "no" in the summary.

I also don't think that the build should work with a build directory. They are designed to work with installed libraries. Using build directories is depending on the internals of a given library (which is bad). The libs in the build directory are sometimes not even suitable for direct use (because they got re-linked in the install phase).

**Comment by bagder** [comment]
> I expect a hard error if it cannot detect valid OpenSSL installation in the given directory.

I too think that would be an improvement.

**Comment by bagder** [comment]
I believe configure detects a given wrong OpenSSL path since d353af001 shipped in 7.61.0.
